File: LEM_Colas/Colas_Arreglos.java
import java.util.Scanner;

public class Colas_Arreglos {

    private static final int MAXSIZE = 5;
    private static int[] queue = new int[MAXSIZE];
    private static int front = -1;
    private static int rear = -1;
    private static Scanner scanner = new Scanner(System.in);

    public static void insertar() {
        int elemento;
        System.out.print("\nIngrese el elemento que se va a insertar ");
        if (scanner.hasNextInt()) {
            elemento = scanner.nextInt();
        } else {
            System.out.println("\nEntrada inválida.");
            scanner.next();
            return;
        }

        if (rear == MAXSIZE - 1) {
            System.out.println("\nDESBORDAMIENTO (OVERFLOW)\n");
            return;
        }
        if (front == -1 && rear == -1) {
            front = rear = 0;
        } else {
            rear++;
        }

        queue[rear] = elemento;
        System.out.println("\nElemento insertado correctamente.\n");
    }

    public static void eliminar() {
        if (front == -1 || front > rear) {
            System.out.println("\nSUBDESBORDAMIENTO (UNDERFLOW)\n");
            return;
        }

        int elemento = queue[front];
        if (front == rear) {
            front = rear = -1;
        } else {
            front++;
        }

        System.out.println("\nElemento correctamente eliminado: " + elemento + "\n");
    }

    public static void mostrar() {
        if (rear == -1 || front == -1 || front > rear) {
            System.out.println("\nLa cola está vacía.\n");
        } else {
            System.out.println("\nElementos en la cola:");
            for (int i = front; i <= rear; i++) {
                System.out.println(queue[i]);
            }
        }
    }

    public static void main(String[] args) {
        int opcion = 0;

        while (opcion != 4) {
            System.out.println("\n========== MENÚ PRINCIPAL ==========");
            System.out.println("1. Insertar un elemento");
            System.out.println("2. Eliminar un elemento");
            System.out.println("3. Mostrar la cola");
            System.out.println("4. Salir");
            System.out.print("Ingrese su opción: ");
            
            if (scanner.hasNextInt()) {
                opcion = scanner.nextInt();
                switch (opcion) {
                    case 1: insertar(); break;
                    case 2: eliminar(); break;
                    case 3: mostrar(); break;
                    case 4: System.out.println("\nCerrando...\n"); break;
                    default: System.out.println("\nOpción inválida.\n"); break;
                }
            } else {
                System.out.println("\nOpción inválida.\n");
                scanner.next();
            }
        }
        scanner.close();
    }
}

File: LEM_Colas/Colas_Arreglos.py
MAXSIZE = 5
queue = [0] * MAXSIZE
front = -1
rear = -1

def insertar():
    global front, rear, queue
    try:
        elemento = int(input("\nIngrese el elemento que se va a insertar "))
    except ValueError:
        print("\nEntrada inválida.")
        return

    if rear == MAXSIZE - 1:
        print("\nDESBORDAMIENTO (OVERFLOW)\n")
        return
    if front == -1 and rear == -1:
        front = rear = 0
    else:
        rear += 1

    queue[rear] = elemento
    print("\nElemento insertado correctamente.\n")

def eliminar():
    global front, rear, queue
    if front == -1 or front > rear:
        print("\nSUBDESBORDAMIENTO (UNDERFLOW)\n")
        return

    elemento = queue[front]
    if front == rear:
        front = rear = -1
    else:
        front += 1

    print(f"\nElemento correctamente eliminado: {elemento}\n")

def mostrar():
    global front, rear, queue
    if rear == -1 or front == -1 or front > rear:
        print("\nLa cola está vacía.\n")
    else:
        print("\nElementos en la cola:")
        for i in range(front, rear + 1):
            print(queue[i])

if __name__ == "__main__":
    opcion = 0

    while opcion != 4:
        print("\n========== MENÚ PRINCIPAL ==========")
        print("1. Insertar un elemento")
        print("2. Eliminar un elemento")
        print("3. Mostrar la cola")
        print("4. Salir")
        
        try:
            opcion = int(input("Ingrese su opción: "))
        except ValueError:
            opcion = 0

        if opcion == 1: insertar()
        elif opcion == 2: eliminar()
        elif opcion == 3: mostrar()
        elif opcion == 4: print("\nCerrando...\n")
        else: print("\nOpción inválida.\n")